### ISSUE TITLE
Make it possible to configure the html element and its class in the output for the footnotes section label

### ISSUE THREAD

**Issue by gtnbssn** [syntax-tree/mdast-util-to-hast#63]
### Initial checklist

- [X] I read the [support docs](https://github.com/syntax-tree/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/syntax-tree/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/syntax-tree/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Asyntax-tree&type=Issues) and couldn’t find anything (or linked relevant results below)

### Problem

The site i am building uses markdown content which has a number of headings structuring it.

It is an academic site in which the title for the footnotes section will be changed to "sources", as all of the notes are sources quoted in the article. I want this title to be accessible to all users.

There are two things i am trying to achieve that would be possible with more options:

- i would like the element enclosing the footnotes title to be an h4 rather than an h2 in order to fit my document's structure better. I am also using tailwind typography to format the content, so being able to choose the html element will allow me to make sure it looks as it should according to this hierarchy without extra css.
- i would also like this element to actually be visible to sighted users. Currently the `sr-only` class is always added if i am not mistaken.

### Solution

Add extra configuration options.

I believe the default behaviour should stay the same.

A `footnoteLabelElement` option would allow to set the html element that will be used. It would default to `h2`.

A `footnoteLabelSROnly` option would allow to add or not the class. It would be true by default.

### Alternatives

Add a "sources" title at the end of the markdown document. This allows me to control with which element it will be rendered. However this means screen readers will see this title, and then the one generated by this tool.

Do not add the `sr-only` class in my project's css. In this case i cannot use this class elsewhere in my project.

I hope this makes sense! Admittedly this is not a huge problem, but something related to my specific use case.

**Comment by gtnbssn** [comment]
Thanks for the quick answer! I have looked into the links you provided and am not sure i understand.

To clarify i am not trying to handle each footnote, but the footer which wraps all of them at the bottom of the content.

This is handled here: https://github.com/syntax-tree/mdast-util-to-hast/blob/dfd724a5e62fc270e71bc2d5a2e4471be0c5ef5b/lib/footer.js#L104-L125

My impression is that this is not a handler because this does not operate on a node from the mardown tree, but rather will wrap around the output of a number of nodes. It is quite possible i am not understanding this properly however.

Is this indeed something i can do with handlers?

**Comment by ChristianMurphy** [comment]
It may make sense to have an option the allows replacing all that hast.
Somewhat similar to https://github.com/rehypejs/rehype-autolink-headings#optionsgroup

But I'd be interested to hear what other maintainers think.

**Comment by wooorm** [comment]
I believe Christian’s suggestion was slightly off, as you indeed are not interested in something that can be achieved with handlers. But the spirit of that idea: changing things, is very much possible with unified and making your own plugins to change things.

Given that you can pass `options.footnoteLabel` already to change the text of the section to `Sources`, what rests is changing the tag name of the heading and removing its class. You could make a small plugin to do what you want:

```js
import { select } from 'hast-util-select'

/** @type {import('unified').Plugin<[], import('hast').Root>} */
export default function myRehypePluginToChangeTheFootnoteSection() {
  return function (tree) {
    const heading = select('section.footnotes h2.sr-only', tree)
    heading.tagName = 'h4' // Change tag name.
    heading.properties.className = undefined // Remove class prop.
  }
}
```

The benefit of using a plugin to serve your needs is that it can exactly match what you want. It sounds like you may want much more customization anyway to make an academic site?

---

We could add options for your original question here. I would instead prefer:

*	`options.footnoteLabelTagName: string` (default `h2`)
*	`options.footnoteLabelProperties: Properties` (default `{className: ['sr-only']}`)

Is this something you are interested in working on?

**Comment by gtnbssn** [comment]
I am interested in giving it a shot! I have something that works and will open a PR.

**Comment by wooorm** [comment]
Released in [`12.2.0`](https://github.com/syntax-tree/mdast-util-to-hast/releases/tag/12.2.0)!
